core/src/main/java/com/airbnb/aerosolve/core/models/FullRankLinearModel.java

```diff
@@ -0,0 +1,164 @@
+package com.airbnb.aerosolve.core.models;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.*;
+
+import com.airbnb.aerosolve.core.DebugScoreRecord;
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.LabelDictionaryEntry;
+import com.airbnb.aerosolve.core.ModelHeader;
+import com.airbnb.aerosolve.core.ModelRecord;
+import com.airbnb.aerosolve.core.MulticlassScoringResult;
+import com.airbnb.aerosolve.core.util.Util;
+import com.airbnb.aerosolve.core.util.FloatVector;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// A full rank linear model that supports multi-class classificaiton.
+// The class vector Y = W' X where X is the feature vector.
+public class FullRankLinearModel extends AbstractModel {
+
+  private static final long serialVersionUID = -849900702679383420L;
+
+  @Getter @Setter
+  private Map<String, Map<String, FloatVector>> weightVector;
+  
+  @Getter @Setter
+  private ArrayList<LabelDictionaryEntry> labelDictionary;
+
+  @Getter @Setter
+  private Map<String, Integer> labelToIndex;
+
+  public FullRankLinearModel() {
+    weightVector = new HashMap<>();
+    labelDictionary = new ArrayList<>();
+  }
+
+  // In the binary case this is just the score for class 0.
+  @Override
+  public float scoreItem(FeatureVector combinedItem) {
+    Map<String, Map<String, Double>> flatFeatures = Util.flattenFeature(combinedItem);
+    FloatVector sum = scoreFlatFeature(flatFeatures);
+    return sum.values[0];
```

Comment:
for multiclass case this is also the score for class 0, to avoid misuse of this method, perhaps add a warning if there are more than two classes?